client/dashboard/plugins/manage/components/plugins-header-actions.tsx

```diff
@@ -0,0 +1,48 @@
+import { Button, __experimentalText as Text } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+
+import './plugins-header-actions.scss';
+
+type PluginsHeaderActionsProps = {
+	updateCount: number;
+	onFilterUpdates: () => void;
+	/**
+	 * Whether the header is rendered in the "sites with this plugin" dataview.
+	 * When true, we use a singular "Plugin up to date" message instead of
+	 * the default plural wording.
+	 */
+	isSitesWithThisPluginView?: boolean;
+};
+
+export const PluginsHeaderActions = ( {
+	updateCount,
+	onFilterUpdates,
+	isSitesWithThisPluginView = false,
+}: PluginsHeaderActionsProps ) => {
+	const hasUpdates = updateCount > 0;
+
+	if ( hasUpdates ) {
+		return (
+			<Button
+				variant="tertiary"
+				size="compact"
+				className="plugins-header-actions__updates-link"
+				onClick={ onFilterUpdates }
```

Comment:
When `isSitesWithThisPluginView` is true, I think we should render this as plain text, not button. Otherwise, this will add a filter which does not exist in the DataViews.